src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationIndexValidator.cs

```diff
@@ -18,34 +18,54 @@ public abstract class RegistrationIndexValidator : RegistrationValidator
         {
         }
 
-        protected override async Task<bool> ShouldRun(ValidationContext data)
+        protected override async Task<bool> ShouldRunAsync(ValidationContext context)
         {
-            return 
-                await base.ShouldRun(data) &&
+            var shouldRunTask = context.GetCachedResultAsync(
```

Comment:
If you're going to cache the result of the base `Validator.ShouldRunAsync`, you should do it in the base `Validator.ShouldRunAsync` method itself. Same applies to `RegistrationValidator.GetIndexAsync` and `RegistrationValidator.GetLeafAsync` and even `RegistrationIndexValidator.ShouldRunAsync` itself